src/searcher.rs

```diff
@@ -247,9 +248,12 @@ impl Searcher {
     ///
     /// Returns the Document, raises ValueError if the document can't be found.
     fn doc(&self, doc_address: &DocAddress) -> PyResult<Document> {
-        let doc = self.inner.doc(doc_address.into()).map_err(to_pyerr)?;
-        let named_doc = self.inner.schema().to_named_doc(&doc);
-        Ok(Document {
+        // Bring the trait into scope
+        use tantivy::Document;
```

Comment:
The term `Document` is overloaded unfortunately. This import is needed to bring the trait into scope so that we can call `to_named_doc` on the `TantivyDocument` struct instance.  And then we also return a different `Document` struct, local to tantivy-py.  Need to fix the imports.

Using the same names between tantivy-py and tantivy is generally confusing, and this is a specific example where we have a namespace collision between the trait and a different struct called `Document`.

Reply:
Note that you can bring a trait into scope without polluting the namespace, e.g.

```rust
use tantitvy::Document as _;
```

Reply:
Nice TIL 💯